exp/addons/internal/controllers/clusterresourcesetbinding_controller.go

```diff
@@ -79,33 +79,35 @@ func (r *ClusterResourceSetBindingReconciler) Reconcile(ctx context.Context, req
 		// Error reading the object - requeue the request.
 		return ctrl.Result{}, err
 	}
+	patchHelper, err := patch.NewHelper(binding, r.Client)
+	if err != nil {
+		return ctrl.Result{}, errors.Wrap(err, "failed to configure the patch helper")
+	}
+	defer func() {
+		if err = patchHelper.Patch(ctx, binding); err != nil {
+			reterr = errors.Wrap(err, "failed to patch ClusterResourceSetBinding to add spec.clusterName")
```

Comment:
```suggestion
			reterr = errors.Wrap(err, "failed to patch ClusterResourceSetBinding")
```
to avoid a misleading error if we change something else